Sources/DockProgress/DockProgress.swift

```diff
@@ -1,13 +1,34 @@
 import Cocoa
+import CoreVideo
 
 @MainActor
 public enum DockProgress {
-	private static var previousProgress: Double = 0
+    private static var t: Double = 0;
+    private static var displayLink: CVDisplayLink?
 	private static var progressObserver: NSKeyValueObservation?
 	private static var finishedObserver: NSKeyValueObservation?
 
 	private static let dockContentView = with(ContentView()) {
 		NSApp.dockTile.contentView = $0
+        
+        let displayLinkCallback: CVDisplayLinkOutputCallback = {(displayLink: CVDisplayLink, inNow: UnsafePointer<CVTimeStamp>, inOutputTime: UnsafePointer<CVTimeStamp>, flagsIn: CVOptionFlags, flagsOut: UnsafeMutablePointer<CVOptionFlags>, displayLinkContext: UnsafeMutableRawPointer?) -> CVReturn in
+            Task { @MainActor in
+                let speed = 1.0
+                t += speed * CVDisplayLinkGetActualOutputVideoRefreshPeriod(displayLink)
+                if (animatedProgress - progress).magnitude <= 0.01 {
+                    animatedProgress = progress
+                    t = 0
+                } else {
+                    animatedProgress = lerp(animatedProgress, progress, easeInOut(t));
```

Comment:
What do you think about making the transition even longer? Feels a bit smoother. (Requires the other change)

```suggestion
                    animatedProgress = Easing.lerp(animatedProgress, progress, Easing.easeInOut(t, 2));
```

Reply:
I'll give it a try!

Reply:
Honestly I don't like this. I think it's better without, at least at the progress rates I've tried. But I will happily oblige if you prefer it :)

Reply:
Alright, we can do without.